Dockerfile

```diff
@@ -1,43 +1,37 @@
 # Build stage
 ARG goversion
 FROM golang:${goversion}-alpine as builder
-RUN apk add build-base git mercurial
-ADD go.mod /spire/go.mod
-RUN cd /spire && go mod download
-ADD . /spire
 WORKDIR /spire
+RUN apk --no-cache --update add build-base git mercurial
+ADD go.* ./
+RUN go mod download
+ADD . .
 RUN make build
 
 # Common base
 FROM alpine AS spire-base
-RUN apk --no-cache add dumb-init
-RUN apk --no-cache add ca-certificates
+WORKDIR /opt/spire
 RUN mkdir -p /opt/spire/bin
+CMD []
+RUN apk --no-cache --update add dumb-init
+RUN apk --no-cache --update add ca-certificates
```

Comment:
Since we're using --no-cache, would it be better to add both dumb-init and ca-certificates in the same invocation?

Reply:
I assume dumb-init doesn't update as often as ca-certificates, so we could benefit from reusing a layer in some builds. Both OK by me, tell me your preference.